Steps4Impact/DashboardV2/DashboardDataSource.swift

```diff
@@ -116,4 +113,68 @@ class DashboardDataSource: TableViewDataSource {
         context: DashboardContext.inviteSupporters)
     ]]
   }
+
+  private var activityCell: CellContext {
+    if UserInfo.pedometerSource != nil {
+      return ActivityCardCellContext(title: Strings.Dashboard.Activity.title,
+                                     milesDayCount: milesCountDay,
+                                     milesWeekCount: milesCountWeek,
+                                     commitment: commitment)
+    } else {
+      return EmptyActivityCellContext(title: Strings.Dashboard.Activity.title,
+                                      body: Strings.Dashboard.Activity.disconnected,
+                                      ctaTitle: Strings.Dashboard.Activity.connect)
+    }
+  }
+
+  private func getStepCounts(completion: @escaping () -> Void) { // swiftlint:disable:this cyclomatic_complexity
+    let now = Date()
+    guard
+      let dayInterval = Calendar.current.dateInterval(of: .day, for: now),
+      let weekInterval = Calendar.current.dateInterval(of: .weekOfYear, for: now)
+    else { return }
+
+    switch UserInfo.pedometerSource {
+    case .healthKit:
+      HealthKitDataProvider().retrieveDistance(forInterval: dayInterval) { [weak self] (result) in
```

Comment:
Might be good to split out the HK and FitBit into separate functions that we call from here.